test-plan.md

```diff
@@ -93,6 +93,38 @@ The easist way to verify that a project was successfully loaded is to open a .cs
 * In a project that uses globbing (.NET Core), use the VS Code file explorer to add a new file next to the csproj. Intellisense/sighelp/etc should be available in the new file
 * Add a new file and reference a type in it from a different file. Deleting from disk the file containing the referenced type  should produce error messages
 
+#### Razor
+The Razor experience is available when you open a .cshtml file in a valid OmniSharp project.
+
+##### C# Completion
+* Typing `@DateTime.Now` and `@(DateTime.Now)` provides completions throughout typing.
+* Completion is available for types that exist in the project (i.e. `Program`)
+* Typing `@model DateTime` prompts for completion for the `model` symbol and the `DateTime` symbol.
+
+#### C# Signature Help
+* Typing `@Html.Raw()` prompts for signature help inside of the `()`.
+
+#### C# Diagnostics
+* Typing `@ThisDoesNotExist` results in an error being created and squiggled in the .cshtml file. NOTE: This error squiggly will be misaligned due to known issues.
+
+#### Razor Options
+
+##### razor.disabled
+This option can be set to `true` to disable the above described C# experience.
+
+##### razor.languageServer.directory
+Setting this path to any of the values as listed below, should start the Razor Language server.
+  * undefined - Razor Language server must start using the copy of Razor Language server shipped with the extension.
+  * Some absolute path to a directory - Razor Language server must start using the local copy of Razor Language server that exists in the path pointed to by the config.
```

Comment:
How does a vendor know what path the razor language server used? Do we print that in the "Razor channel", like "Razor server started...with path ..."

Reply:
Sadly we don't print where the server is being run from. We just print what version is being used (the assembly version of the server). I've filed an issue to add that information: https://github.com/aspnet/Razor.VSCode/issues/169

Reply:
I'd guess the only aspect of `razor.languageServer.directory` that a vendor could verify is that it causes an error if you point to some directory that doesn't exist or doesn't contain `rzls[.exe]`. They won't be able to verify it actually works if you point to a real alternate language server, because they won't have a real alternate language server.

TBH I don't think this is something the vendors really need to verify. The option only exists for people working on Razor Language Server.

Reply:
Alright fair enough, i'll nuke this requirement in the test plan.